### PR TITLE
Reliability docs

### LINKED ISSUE #562: Document reliability package

### PR DESCRIPTION
**Pull request by mantzas**
## Which problem is this PR solving?

Closes #562.

## Short description of the changes

- Improved docs.
- Removed markdown.

### CODE REVIEW

File: reliability/circuitbreaker/breaker.go [hunk 1]
```diff
@@ -51,11 +51,11 @@ func breakerCounterInc(name string, st status) {
 
 // Setting definition.
 type Setting struct {
-	// The threshold for the circuit to open.
+	// The failure threshold for the circuit to open.
```
Comment (pkritiotis) [inline reliability/circuitbreaker/breaker.go]: _Not strong on this at all._

Should we use the strict commenting convention of including the name of the field we are commenting?

Does this affect godoc? I actually thought that this was a requirement.

for example 
```suggestion
	// FailureThreshold represents the failure threshold for the circuit to open.
```

**Review by pkritiotis — DISMISSED** [review]

File: reliability/circuitbreaker/breaker.go (hunk 1 above)
Comment (mantzas) [reply reliability/circuitbreaker/breaker.go]: Nice catch!

**Review by pkritiotis — DISMISSED** [review]

File: reliability/retry/retry_test.go [hunk 2]
```diff
@@ -84,29 +87,32 @@ func Test_Retry_Execute(t *testing.T) {
 			expectErr:          true,
 		},
 	}
-	for name, tC := range testCases {
+	for name, tt := range testCases {
```
Comment (xenosdio) [inline reliability/retry/retry_test.go]: Should we rename `testCases` to `tests`? To follow the convention we use in all other table test vars?

**Review by xenosdio — CHANGES_REQUESTED** [review]

File: reliability/retry/retry_test.go (hunk 2 above)
Comment (mantzas) [reply reliability/retry/retry_test.go]: done

**Review by xenosdio — APPROVED** [review]

**Review by pkritiotis — APPROVED** [review]

**Pull request merged**